src/peerbook/libp2p_peer.erl

```diff
@@ -186,14 +314,35 @@ decode_list(Bin) ->
 -spec decode(binary()) -> peer().
 decode(Bin) ->
     Msg = libp2p_peer_pb:decode_msg(Bin, libp2p_signed_peer_pb),
-    verify(Msg).
+    true = verify(Msg),
+    Msg.
```

Comment:
This function can badmatch now, is this expected?

Reply:
Can't we just let verify() or association_verify() error and let it bubble up?

Reply:
ah ok.. I can ignore the response of verify since dialyzer can't catch them all ahead of time

Reply:
Can or can't?

Reply:
Can.. fixed in later commit